File: we_8_4_one_full_step_of_a_position.m
% Integrated Navigation Systems — Chapter 8: Optimal Estimation and the Kalman Filter
% Worked Example 8.4 — One full step of a position–velocity filter
% Companion code (educational reference implementation; see the book for context).

Pm=[1.0 0.2;0.2 0.5]; H=[1 0]; R=0.25;
S=H*Pm*H'+R; K=Pm*H'/S;
fprintf('K=[%.2f %.2f]\n', K(1), K(2));   % 0.80 0.16

File: we_8_4_one_full_step_of_a_position.py
# Integrated Navigation Systems — Chapter 8: Optimal Estimation and the Kalman Filter
# Worked Example 8.4 — One full step of a position–velocity filter
# Companion code (educational reference implementation; see the book for context).

import numpy as np
Pm=np.array([[1.0,0.2],[0.2,0.5]]); H=np.array([[1,0]]); R=0.25
S=H@Pm@H.T+R; K=Pm@H.T@np.linalg.inv(S)
print('K=', np.round(K.ravel(),2))   # [0.80 0.16]
